### PR TITLE
Support cluster cleanup policy after job finishes

### PR DESCRIPTION
**Pull request by functicons**

### CODE REVIEW

File: controllers/flinkcluster_converter.go [hunk 1]
```diff
@@ -605,18 +586,37 @@ func getJobManagerIngressHost(ingressHostFormat string, clusterName string) stri
...
+	if jobStatus == nil {
 		return false
 	}
-	var tc = &TimeConverter{}
-	var lastUpdateTime = tc.FromString(clusterStatus.LastUpdateTime)
-	return now.After(
-		lastUpdateTime.Add(time.Duration(delayMinutes) * time.Minute))
+
+	// Job hasn't finished yet.
+	if jobStatus.State != v1alpha1.JobState.Succeeded &&
```
Comment (hongyegong) [inline controllers/flinkcluster_converter.go]: This is confused with the method name ...AfterJobFinishes?

File: api/v1alpha1/flinkcluster_types.go [hunk 2]
```diff
@@ -23,17 +23,19 @@ import (
...
-	Creating    string
-	Running     string
-	Reconciling string
-	Stopping    string
-	Stopped     string
+	Creating         string
+	Running          string
+	Reconciling      string
+	Stopping         string
+	PartiallyStopped string
```
Comment (hongyegong) [inline api/v1alpha1/flinkcluster_types.go]: What does this mean given the cluster context?

File: controllers/flinkcluster_converter.go (hunk 1 above)
Comment (functicons) [reply controllers/flinkcluster_converter.go]: Updated.

File: api/v1alpha1/flinkcluster_types.go (hunk 2 above)
Comment (functicons) [reply api/v1alpha1/flinkcluster_types.go]: It corresponds to cleanup action `DeleteTaskManager`.

File: api/v1alpha1/flinkcluster_validate.go [hunk 3]
```diff
@@ -162,6 +162,9 @@ func (v *Validator) validateJob(jobSpec *JobSpec) error {
 	if *jobSpec.Parallelism < 1 {
 		return fmt.Errorf("job parallelism must be >= 1")
 	}
+	if jobSpec.RestartPolicy == nil {
+		return fmt.Errorf("job restartPolicy is unspecified")
+	}
 	if jobSpec.RestartPolicy != nil {
```
Comment (hongyegong) [inline api/v1alpha1/flinkcluster_validate.go]: rm.

File: api/v1alpha1/flinkcluster_validate.go [hunk 4]
```diff
@@ -170,6 +173,22 @@ func (v *Validator) validateJob(jobSpec *JobSpec) error {
 			return fmt.Errorf("invalid job restartPolicy: %v", *jobSpec.RestartPolicy)
 		}
 	}
+	if jobSpec.CleanupPolicy == nil {
+		return fmt.Errorf("job cleanupPolicy is unspecified")
+	}
+	if jobSpec.CleanupPolicy != nil {
```
Comment (hongyegong) [inline api/v1alpha1/flinkcluster_validate.go]: can be rmed.

**Comment by functicons** [comment]
Fix #66

**Pull request merged**